engine/crates/parser-sdl/src/rules/mongodb_directive/model_directive/types/input.rs

```diff
@@ -24,13 +24,15 @@ pub(crate) fn register_type_input(
         let is_list = field.r#type().base.is_list();
         let is_optional = field.r#type().nullable;
 
-        let composite_type = visitor_ctx.types.get(base).filter(|r#type| r#type.is_composite());
+        let r#type = visitor_ctx.types.get(base);
+        let composite_type = r#type.map(|r#type| r#type.is_composite()).unwrap_or_default();
 
-        let r#type = match composite_type {
-            Some(_) if is_list => {
+        let r#type = match (r#type, composite_type) {
+            (Some(r#type), false) if matches!(r#type.kind, engine_parser::types::TypeKind::Enum(_)) => base.to_owned(),
```

Comment:
Is this right? I would assume enums should have the same API as scalars, but we don't create the input types for them.